diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8b23d1..cf82bb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,8 @@
 cmake_minimum_required(VERSION 3.16)
 
-project(QAdwaitaDecorations
+project(QYaruDecorations
     VERSION 0.1.7
-    DESCRIPTION "Qt Wayland decoration plugin using libadwaita style"
+    DESCRIPTION "Qt Wayland decoration plugin using the yaru style"
     LANGUAGES CXX C)
 
 option(USE_QT6 "Use Qt6 instead of Qt5" OFF)
@@ -10,26 +10,26 @@ option(USE_QT6 "Use Qt6 instead of Qt5" OFF)
 set(CMAKE_AUTOMOC ON)
 
 if (USE_QT6)
-    set(QT_MIN_VERSION "6.5.0")
-    set(CMAKE_CXX_STANDARD 17)
-    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(QT_MIN_VERSION "6.5.0")
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
 else()
-    set(QT_MIN_VERSION "5.15.2")
-    set(CMAKE_CXX_STANDARD 14)
-    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(QT_MIN_VERSION "5.15.2")
+  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
 endif()
 
 include(GNUInstallDirs)
 include(FeatureSummary)
 
 if (USE_QT6)
-    find_package(QT NAMES Qt6 COMPONENTS Core Gui Svg Wayland Widgets REQUIRED)
-    if (Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
-        find_package(Qt6GuiPrivate REQUIRED)
-        find_package(Qt6WaylandClientPrivate REQUIRED)
-    endif()
+  find_package(QT NAMES Qt6 COMPONENTS Core Gui Svg Wayland Widgets REQUIRED)
+  if (Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
+    find_package(Qt6GuiPrivate REQUIRED)
+    find_package(Qt6WaylandClientPrivate REQUIRED)
+  endif()
 else()
-    find_package(QT NAMES Qt5 COMPONENTS Core Gui Svg Wayland Widgets REQUIRED)
+  find_package(QT NAMES Qt5 COMPONENTS Core Gui Svg Wayland Widgets REQUIRED)
 endif()
 
 find_package(Qt${QT_VERSION_MAJOR} ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
@@ -40,7 +40,7 @@ find_package(Qt${QT_VERSION_MAJOR} ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
     Widgets
 )
 if (Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
-    find_package(Qt6 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
+  find_package(Qt6 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
         GuiPrivate
         WaylandClientPrivate
     )
@@ -48,31 +48,31 @@ endif()
 
 find_package(Qt${QT_VERSION_MAJOR}Gui ${QT_MIN_VERSION} CONFIG REQUIRED Private)
 if (NOT USE_QT6)
-    find_package(Qt${QT_VERSION_MAJOR}ThemeSupport REQUIRED)
+  find_package(Qt${QT_VERSION_MAJOR}ThemeSupport REQUIRED)
 endif()
 
 # NOTE: I don't know how to do this only in case of qt_config(xkbcommon).
 # We would miss an include in QWaylandDisplay header file.
 if (NOT USE_QT6)
-    find_package(Qt${QT_VERSION_MAJOR}XkbCommonSupport ${QT_MIN_VERSION})
+  find_package(Qt${QT_VERSION_MAJOR}XkbCommonSupport ${QT_MIN_VERSION})
 endif()
 
 if (HAS_QT6_SUPPORT OR USE_QT6)
-    message(STATUS "Enabling Qt Wayland decoration shadows support")
-    message(STATUS "NOTE: This support requires changes in Qt Wayland from Qt 6.2")
-    add_definitions(-DHAS_QT6_SUPPORT)
+  message(STATUS "Enabling Qt Wayland decoration shadows support")
+  message(STATUS "NOTE: This support requires changes in Qt Wayland from Qt 6.2")
+  add_definitions(-DHAS_QT6_SUPPORT)
 endif()
 
 if (NOT QT_PLUGINS_DIR)
-    if (NOT USE_QT6)
-        get_target_property(REAL_QMAKE_EXECUTABLE ${Qt5Core_QMAKE_EXECUTABLE}
+  if (NOT USE_QT6)
+    get_target_property(REAL_QMAKE_EXECUTABLE ${Qt5Core_QMAKE_EXECUTABLE}
                             IMPORTED_LOCATION)
-        execute_process(COMMAND "${REAL_QMAKE_EXECUTABLE}" -query QT_INSTALL_PLUGINS
+    execute_process(COMMAND "${REAL_QMAKE_EXECUTABLE}" -query QT_INSTALL_PLUGINS
                         OUTPUT_VARIABLE QT_PLUGINS_DIR
                         ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
-    else()
-        set(QT_PLUGINS_DIR ${QT6_INSTALL_PLUGINS})
-    endif()
+  else()
+    set(QT_PLUGINS_DIR ${QT6_INSTALL_PLUGINS})
+  endif()
 endif()
 
 add_subdirectory(src)
diff --git a/README.md b/README.md
index 8f045cf..43844f6 100644
--- a/README.md
+++ b/README.md
@@ -1,7 +1,9 @@
-# QAdwaitaDecorations
-Qt decoration plugin implementing Adwaita-like client-side decorations.
+# QYaruDecorations
+
+Qt decoration plugin implementing a Yaru-like client-side decorations.
 
 ## How to compile
+
 This library uses private Qt headers and will likely not be forward nor
 backward compatible. This library will have to be recompiled with every
 Qt update. While it can be build using Qt 5, it is recommended to get
@@ -17,17 +19,34 @@ make && make install
 ```
 
 ## Usage
+
 It can be used by setting the QT_WAYLAND_DECORATION environment variable:
 
 ```
-export QT_WAYLAND_DECORATION=adwaita
+export QT_WAYLAND_DECORATION=yaru
 ```
 
-Newer version of Qt6 includes an implementation of this plugin. To force decorations from this plugin, set the QT_WAYLAND_DECORATION environment variable to the following value:
+or
+
 ```
-export QT_WAYLAND_DECORATION=qadwaitadecorations
+export QT_WAYLAND_DECORATION=qyarudecorations
+```
+
+### NOTE:
+
+if you want this to persiste and apply to all system Qt applications, use /etc/environment.d
+if that directory doesn't already exist create it with
+
+```
+sudo mkdir /etc/environment.d
+```
+
+then create a generic numbered file there containing the enviroment variable
+
+```
+echo "QT_WAYLAND_DECORATIONS=yaru" | sudo tee /etc/environment.d/10-qt-yaru-decorations
 ```
 
 ## License
-The code is under [LGPL 2.1](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html) with the "or any later version" clause.
 
+The code is under [LGPL 2.1](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html) with the "or any later version" clause.
diff --git a/src/qadwaitadecorations.cpp b/src/qadwaitadecorations.cpp
index 8b2abc2..b93a728 100644
--- a/src/qadwaitadecorations.cpp
+++ b/src/qadwaitadecorations.cpp
@@ -17,7 +17,7 @@
  *
  */
 
-#include "qadwaitadecorations.h"
+#include "qyarudecorations.h"
 
 #include <QtWaylandClient/private/qwaylandshellsurface_p.h>
 #include <QtWaylandClient/private/qwaylandshmbackingstore_p.h>
diff --git a/src/qadwaitadecorations.h b/src/qadwaitadecorations.h
index ba47406..72991b2 100644
--- a/src/qadwaitadecorations.h
+++ b/src/qadwaitadecorations.h
@@ -17,8 +17,8 @@
  *
  */
 
-#ifndef QADWAITA_DECORATIONS_H
-#define QADWAITA_DECORATIONS_H
+#ifndef QYARU_DECORATIONS_H
+#define QYARU_DECORATIONS_H
 
 #include <QtCore/QDateTime>
 #include <QtGui/QPixmap>
@@ -32,7 +32,7 @@ using namespace QtWaylandClient;
 class QDBusVariant;
 class QPainter;
 
-class QAdwaitaDecorations : public QWaylandAbstractDecoration
+class QYaruDecorations : public QWaylandAbstractDecoration
 {
     Q_OBJECT
 public:
@@ -53,8 +53,8 @@ public:
     Q_DECLARE_FLAGS(Buttons, Button);
     enum ButtonIcon { CloseIcon, MinimizeIcon, MaximizeIcon, RestoreIcon };
 
-    QAdwaitaDecorations();
-    virtual ~QAdwaitaDecorations() = default;
+    QYaruDecorations();
+    virtual ~QYaruDecorations() = default;
 
 protected:
 #ifdef HAS_QT6_SUPPORT
@@ -118,6 +118,6 @@ private:
     QMap<ButtonIcon, QString> m_icons;
 };
 
-Q_DECLARE_OPERATORS_FOR_FLAGS(QAdwaitaDecorations::Buttons)
+Q_DECLARE_OPERATORS_FOR_FLAGS(QYaruDecorations::Buttons)
 
-#endif // QADWAITA_DECORATIONS_H
+#endif // QYARU_DECORATIONS_H
diff --git a/src/qadwaitadecorations.json b/src/qadwaitadecorations.json
index 62d8027..dea58f1 100644
--- a/src/qadwaitadecorations.json
+++ b/src/qadwaitadecorations.json
@@ -1,5 +1,3 @@
 {
-    "Keys": [ "adwaita", "gnome", "qadwaitadecorations" ]
+  "Keys": ["yaru", "gnome", "qyarudecorations"]
 }
-
-
diff --git a/src/qadwaitadecorationsplugin.cpp b/src/qadwaitadecorationsplugin.cpp
index 07c152f..63c9098 100644
--- a/src/qadwaitadecorationsplugin.cpp
+++ b/src/qadwaitadecorationsplugin.cpp
@@ -17,14 +17,14 @@
  *
  */
 
-#include "qadwaitadecorationsplugin.h"
-#include "qadwaitadecorations.h"
+#include "qyarudecorationsplugin.h"
+#include "qyarudecorations.h"
 
-QWaylandAbstractDecoration *QAdwaitaDecorationsPlugin::create(const QString &system,
-                                                              const QStringList &paramList)
+QWaylandAbstractDecoration *QYaruDecorationsPlugin::create(const QString &system,
+                                                           const QStringList &paramList)
 {
     Q_UNUSED(paramList)
-    if (system == "adwaita" || system == "gnome" || system == "qadwaitadecorations")
-        return new QAdwaitaDecorations();
+    if (system == "yaru" || system == "gnome" || system == "qyarudecorations")
+        return new QYaruDecorations();
     return nullptr;
 }
diff --git a/src/qadwaitadecorationsplugin.h b/src/qadwaitadecorationsplugin.h
index 749ac82..780d9d4 100644
--- a/src/qadwaitadecorationsplugin.h
+++ b/src/qadwaitadecorationsplugin.h
@@ -17,20 +17,20 @@
  *
  */
 
-#ifndef QADWAITA_DECORATIONS_PLUGIN_H
-#define QADWAITA_DECORATIONS_PLUGIN_H
+#ifndef QYARU_DECORATIONS_PLUGIN_H
+#define QYARU_DECORATIONS_PLUGIN_H
 
 #include <QtWaylandClient/private/qwaylanddecorationplugin_p.h>
 
 using namespace QtWaylandClient;
 
-class QAdwaitaDecorationsPlugin : public QWaylandDecorationPlugin
+class QYaruDecorationsPlugin : public QWaylandDecorationPlugin
 {
     Q_OBJECT
-    Q_PLUGIN_METADATA(IID QWaylandDecorationFactoryInterface_iid FILE "qadwaitadecorations.json")
+    Q_PLUGIN_METADATA(IID QWaylandDecorationFactoryInterface_iid FILE "qyarudecorations.json")
 public:
     QWaylandAbstractDecoration *create(const QString &system,
                                        const QStringList &paramList) override;
 };
 
-#endif // QADWAITA_DECORATIONS_PLUGIN_H
+#endif // QYARU_DECORATIONS_PLUGIN_H
